Feature/user authentication

kid-sid opened the pull request:
Check the code (backend) and let me know if any change is required. For faster review, you can go through the test cases in the testing folder. The auth process is not yet integrated with the frontend, but the backend API can store and retrieve data from supabase.

kid-sid committed:
Examples for .env variables.

kid-sid committed:
Added model specific variables with their respective values.

kid-sid committed:
Added loggers and changed the port to 8001

kid-sid committed:
Added project related info for the developers.

kid-sid committed:
Added the changes and also the overview of the architecture.

kid-sid committed:
Added supabse for backend.

kid-sid committed:
Added model specific variables (LAZY).

kid-sid committed:
Added model specific variables (PRO).

kid-sid committed:
Added test cases for multiple backend components.

kid-sid committed:
Added auth router.

kid-sid committed:
Added caching and model specific variables.

kid-sid committed:
Added auth schema for supabase.

kid-sid committed:
Added validation conditions for inference request and response.

kid-sid committed:
Created auth service using supabase as the backend.

kid-sid committed:
Changed the code to handle different types of error and also added the rate limiter.

kid-sid committed:
Enhanced logging setup with additional parameters, improved error handling for OpenAI API, and added utility functions for prompt validation and sanitization.

kid-sid committed:
Added a debug statement for refresh token.

kid-sid committed:
Added some folders and files.

kid-sid committed:
Updated README.md to include a complete frontend authentication UI, featuring user registration, login, and logout functionalities. Adjusted API documentation for signup and logout processes, and added a changelog for version 2.1.0. Enhanced access URLs and file structure details.

kid-sid committed:
Added endpoint to serve the authentication page at /auth, returning the static auth.html file.

kid-sid committed:
Refactored authentication routes: renamed /register to /signup, improved logout logging with token details, made health check timestamp dynamic, and added endpoint to create missing user profiles in the database.

kid-sid committed:
Refactored auth_schema.py by removing unused imports and cleaning up whitespace for improved readability.

kid-sid committed:
Enhanced AuthService with improved profile creation and verification, added diagnostics for profiles table, and implemented error handling for missing profiles. Included detailed logging for table access and structure checks.

kid-sid committed:
Add frontend authentication functionality with signup, login, and logout features. Implemented error handling and user feedback for each action. Tokens are stored in localStorage for session management.

kid-sid committed:
Added static HTML file for authentication page, including forms for signup, login, and logout functionalities. Styled the page for better user experience and integrated JavaScript for handling form submissions.

kid-sid committed:
Implemented authentication UI enhancements in index.html, including user login/logout sections, improved styling for prompts, and added JavaScript functionality for token validation and session management. Updated CSS variables for better theming and user feedback.

kid-sid committed:
Enhanced authentication UI in index.html by adding user login/logout sections, improved styling, and JavaScript functionality for token validation and session management. Updated CSS variables for better theming and user feedback.

pinku3019 commented on routes/auth_router.py:
```diff
@@ -0,0 +1,205 @@
+
+# Dependency to get current user from token
+async def get_current_user(credentials: HTTPAuthorizationCredentials = Depends(security)) -> UserProfile:
+    """Get current authenticated user from JWT token"""
```
add proper docstrings with details like arguments, returns, raises.

kid-sid replied:
Added.

pinku3019 commented on routes/auth_router.py:
```diff
@@ -0,0 +1,205 @@
+
+@router.post("/signup", response_model=UserRegisterResponse, status_code=status.HTTP_201_CREATED)
+async def register_user(user_data: UserRegisterRequest):
+    """Register a new user with email and password"""
```
for all the functions, add docstings with arguments, returns and raises details

kid-sid replied:
Added

pinku3019 commented:
tests are passing as expected.

all 10/10 tests

pinku3019 commented:
the E2E testing failed for Authentication.
its probably due to auth is not implemented.

pinku3019 commented:
make the code to pylint standards  for a organised and cleaner code.
copilot and cursor mostly missing that things.

pinku3019 commented:
for auto pylint checks.
create a file with below contents`
name: Pylint

on: [push]

jobs:
  build:
    runs-on: ubuntu-latest
    strategy:
      matrix:
        python-version: ["3.8", "3.9", "3.10"]
    steps:
    - uses: actions/checkout@v4
    - name: Set up Python ${{ matrix.python-version }}
      uses: actions/setup-python@v3
      with:
        python-version: ${{ matrix.python-version }}
    - name: Install dependencies
      run: |
        python -m pip install --upgrade pip
        pip install pylint
    - name: Analysing the code with pylint
      run: |
        pylint $(git ls-files '*.py') --disable=R`
 name: `reprompt/.github/workflows/pylint.yml`

kid-sid committed:
Add .pylintrc configuration file for Python linting setup.

kid-sid committed:
Updated frontend routes to serve chatbot.html and auth.html from static.

kid-sid committed:
Updated PROJECT_STATUS.md to reflect version 2.1.0, enhancing the project status with comprehensive documentation and new frontend features including a modern responsive interface, authentication UI, and real-time chat. Added details on testing and code quality improvements.

kid-sid committed:
Refactor README.md to enhance project description and features overview. Updated architecture diagram, core features, and quick start instructions. Improved API documentation for authentication and inference endpoints, and added testing and deployment sections for better clarity.

kid-sid committed:
Update requirements.txt to include 'supabase' and 'pylint' for enhanced project dependencies and linting support.

kid-sid committed:
Added customModel class in load_model.py for future model implementation.

kid-sid committed:
Enhanced authentication router documentation in auth_router.py. Added comprehensive docstrings for all endpoints, detailing parameters, return types, and error handling. Included module-level documentation summarizing functionality and dependencies for better clarity and maintainability.

kid-sid committed:
Enhanced documentation for inference router endpoints in inference_router.py. Added detailed docstrings for each endpoint, including descriptions, parameters, return types, error handling, and usage notes to improve clarity and maintainability.

kid-sid committed:
Add authentication UI in auth.html with login and signup forms, including responsive styling and JavaScript functionality for form handling.

kid-sid committed:
Refactored auth.js to improve authentication handling with modular functions for signup, login, and logout. Streamlined API calls using constants from config.js. Added form visibility toggling and session checks on page load.

kid-sid committed:
Added chatbot.html for the Reprompt Chatbot interface, featuring a responsive design, theme toggle, user authentication section, and prompt optimization form. Includes styling and layout for improved user experience.

kid-sid committed:
Added chatbot.js to implement user authentication, logout functionality, theme and inference toggles, and prompt optimization handling. Includes UI initialization and token validation on page load for enhanced user experience.

kid-sid committed:
Added config.js to define API endpoints, storage keys, and routing constants for authentication and chatbot functionality.

kid-sid committed:
Removed index.html file containing the Reprompt Chatbot interface, including UI elements as it was noot required anymore.

kid-sid committed:
Removed new_auth.html file containing the authentication UI, including signup, login, and logout forms, as it is no longer needed.

kid-sid committed:
Removed auth_test_ui.html file containing the authentication testing UI, including forms for user registration, login, and logout, as it is no longer needed.

kid-sid committed:
No longer required

kid-sid committed:
Duplicate file.

kid-sid committed:
Added comprehensive test suite for Reprompt Chatbot API

This new file includes tests for all API endpoints, Supabase connection, and overall system functionality. It features logging for various test outcomes and a structured approach to validate server responses, user authentication, and error handling.

kid-sid commented:
Made changes. Added the complete auth process. SignIn, LogIn, LogOut with proper JWT.
Added a .pylintrc
